### PR TITLE
Replace `openai-edge` dependency with `openai` v4

### PR DESCRIPTION
**Pull request by petersalas**
- Replace dependency
- Lazily instantiate OpenAI (and Anthropic, for symmetry) clients (OpenAI client initialization fails with an exception if the environment variable is not present)
- Remove OpenAI SSE parsing logic for streams
- (Unrelated) add `outputs` to `turbo.json` so that it can properly restore outputs from the cache

### CODE REVIEW

File: packages/ai-jsx/src/lib/util.ts [hunk 1]
```diff
@@ -8,12 +8,22 @@ export function getEnvVar(name: string, shouldThrow: boolean = true) {
...
+  let env = globalThis.process?.env ?? undefined;
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (env === undefined) {
+    // Webpack for Create React App requires `process.env`, but in other environments
+    // process might not be defined, so we need to try/catch it.
+    try {
+      env = process.env;
+    } catch {
+      // Do nothing, leave it undefined.
+    }
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/util.ts:20]: Is there a reason we can't just do a `'env' in process` check? The `try/catch` feels a bit awkward.

File: packages/ai-jsx/src/lib/openai.tsx [hunk 2]
```diff
@@ -115,42 +100,6 @@ export function OpenAI({
   return result;
 }
 
-export const SSE_PREFIX = 'data: ';
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/openai.tsx:118]: Damn it's nice if we can yank all this.

File: turbo.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
       "dependsOn": ["^build"]
     },
     "build": {
-      "dependsOn": ["^build"]
+      "dependsOn": ["^build"],
+      "outputs": ["dist/**", ".next/**", "!.next/cache/**"]
```
Comment (NickHeiner) [inline turbo.json:9]: heh I knew we weren't actually using Turbo very well – neat if this is the thing that makes it work.

**Review by NickHeiner — APPROVED** [review]

File: packages/ai-jsx/src/lib/util.ts (hunk 1 above)
Comment (petersalas) [reply packages/ai-jsx/src/lib/util.ts:20]: The tricky part is:
- webpack apparently needs to see the literal `process.env` to replace it with the build-time value
- some environments don't have `process` defined at all

...so this was the only way I could figure out how to do it.

**Pull request merged**